offload_vio/server_tx/plugin.cpp

```diff
@@ -0,0 +1,121 @@
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+#include "common/phonebook.hpp"
+
+#include <ecal/ecal.h>
+#include <ecal/msg/protobuf/publisher.h>
+
+#include "vio_output.pb.h"
+
+using namespace ILLIXR;
+
+class server_writer : public plugin {
+public:
+    server_writer(std::string name_, phonebook* pb_)
+		: plugin{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+		, _m_imu_int_input{sb->get_reader<imu_integrator_input>("imu_integrator_input")}
+    { 
+		// Initialize eCAL and create a protobuf publisher
+		eCAL::Initialize(0, NULL, "VIO Offloading Sensor Data Writer");
+		publisher = eCAL::protobuf::CPublisher
+		<vio_output_proto::VIOOutput>("vio_output");
+	}
+
+
+    virtual void start() override {
+        plugin::start();
+
+        sb->schedule<pose_type>(id, "slow_pose", [this](switchboard::ptr<const pose_type> datum, std::size_t) {
+			this->send_vio_output(datum);
+		});
+	}
+
+
+    void send_vio_output(switchboard::ptr<const pose_type> datum) {
+
+		// Construct slow pose for output
+		vio_output_proto::SlowPose* protobuf_slow_pose = new vio_output_proto::SlowPose();
+		protobuf_slow_pose->set_timestamp(datum->sensor_time.time_since_epoch().count());
+
+		vio_output_proto::Vec3* position = new vio_output_proto::Vec3();
```

Comment:
Nit: Is there some API like this
```
vio_output_proto::Vec3* position = new vio_output_proto::Vec3(datum->position.x(), ...y(), ...z());
```
or even
```
vio_output_proto::Vec3* position = new vio_output_proto::Vec3(datum->position);
```

Reply:
Unfortunately the only other constructor the generated file contains is a copy constructor.